src/datapool/datapool_pmem.c

```diff
@@ -121,27 +129,29 @@ datapool_initialize(struct datapool *pool)
     datapool_sync_hdr(pool);
 
     /* 2. fill in the data */
-    pool->hdr->internal.version = DATAPOOL_VERSION;
-    pool->hdr->internal.size = pool->mapped_len;
-    pool->hdr->internal.flags = 0;
+    pool->hdr->version = DATAPOOL_VERSION;
+    pool->hdr->size = pool->mapped_len;
+    pool->hdr->flags = 0;
     datapool_sync_hdr(pool);
 
-    /* 3. set the signature */
-    cc_memcpy(pool->hdr->internal.signature, DATAPOOL_SIGNATURE, DATAPOOL_SIGNATURE_LEN);
+    /* 3. set the signatures */
```

Comment:
Writing of the signature is last because it's used to indicate that writing has finished, i.e., if the signature is written, it means that other metadata must be valid.
You should move the user signature writing to the second step.

Reply:
Done.